Automattic/woocommerce-payments
# Remove AccountOverview namespace and define DepositStatus type

Resolves #5987 

#### Changes proposed in this Pull Request

This PR makes the following changes:

1. Introduce a new `DepositStatus` type defined in `/client/types/deposits.d.ts`. 
2. Update places where deposit statuses were defined but weren't strictly defined (ie just `string`) or were previously defined but not by an accessible type. 
3. Update the `AccountOverview` namespace to use this new `DepositStatus` type in one of its definitions. To do this, I had to remove the namespace. This is ok though because it's [needless-namespacing](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#needless-namespacing) anyway.
4. Because I got rid of the namespace, I then had to update all files that make use of the AccountOverview to now import it. 

#### Testing instructions

Because this PR changes the use of AccountOverview in 6 files you should test that those impacted still work: 

**Balances tab**, **Next deposit**, **Deposit status pill**, **Deposit schedule**
   - Set the wp option `_wcpay_feature_simplify_deposits_ui` to 1 to enable the feature flag.
   - Go to the **Payments > Overview** page
   - Note the Deposit History section heading and deposit schedule.
   - View the Balances Card, Next deposit and Deposit History card. 
   - Confirm all components load as expected

**Deposits Information (legacy balances UI)**
   - Disable the `_wcpay_feature_simplify_deposits_ui` option. 
   - Go to the **Payments > Overview** page
   - View the Balance table.
   - Confirm the balance components load correctly. 


-------------------


**Post merge**
